subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -1079,23 +1080,34 @@ uint32_t ull_adv_aux_start(struct ll_adv_aux_set *aux, uint32_t ticks_anchor,
 {
 	uint32_t volatile ret_cb;
 	uint32_t interval_us;
+	uint32_t ticks_slot;
 	uint8_t aux_handle;
 	uint32_t ret;
 
-	interval_us = aux->interval * PERIODIC_INT_UNIT_US;
-
 	ull_hdr_init(&aux->ull);
 	aux_handle = ull_adv_aux_handle_get(aux);
+	interval_us = aux->interval * PERIODIC_INT_UNIT_US;
+	ticks_slot = aux->ull.ticks_slot + ticks_slot_overhead;
+
+#if defined(CONFIG_BT_CTLR_SCHED_ADVANCED)
+	uint32_t ticks_anchor_aux;
+	int err;
+
+	err = ull_sched_after_adv_aux_slot_get(TICKER_USER_ID_THREAD,
+					       ticks_slot, &ticks_anchor_aux);
+	if (!err) {
```

Comment:
What if there is an error? This it what I would ask if I didn't see the code of the `ull_sched_after_adv_aux_slot_get`. 
The code should not force you to read body every called function to understand what is going on.

I suggest to change the name of the function or do not use `err` as a variable where returned value is assigned if the returned value does not mean an error code.

Reply:
I do not think the caller functions needs to be read, clearly if there is an error then the return parameter is not used. If there is no free slot (or no existing auxiliary sets), the callee uses the calculations made using current tick is used as anchor reference.